tests/components/blue_current/test_config_flow.py

```diff
@@ -0,0 +1,117 @@
+        "bluecurrent_api.Client.get_email", return_value="test@email.com"
+    ), patch(
+        "homeassistant.components.blue_current.async_setup_entry",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "api_token": "123",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["title"] == "test@email.com"
+    assert result2["data"] == {"api_token": "123"}
+
+
+async def test_form_invalid_token(hass: HomeAssistant) -> None:
+    """Test if an invalid api token is handled."""
+    with patch(
+        "bluecurrent_api.Client.validate_api_token",
+        side_effect=InvalidApiToken,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={"api_token": "123"},
+        )
+        assert result["errors"] == {"base": "invalid_token"}
+
+
+async def test_form_limit_reached(hass: HomeAssistant) -> None:
+    """Test if an limit reached error is handled."""
+    with patch(
+        "bluecurrent_api.Client.validate_api_token",
+        side_effect=RequestLimitReached,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={"api_token": "123"},
+        )
+        assert result["errors"] == {"base": "limit_reached"}
+
+
+async def test_form_already_connected(hass: HomeAssistant) -> None:
+    """Test if an already connected error is handled."""
+    with patch(
+        "bluecurrent_api.Client.validate_api_token",
+        side_effect=AlreadyConnected,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={"api_token": "123"},
+        )
+        assert result["errors"] == {"base": "already_connected"}
+
+
+async def test_form_exception(hass: HomeAssistant) -> None:
+    """Test if an exception is handled."""
+    with patch(
+        "bluecurrent_api.Client.validate_api_token",
+        side_effect=Exception,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={"api_token": "123"},
+        )
+        assert result["errors"] == {"base": "unknown"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test if a connection error is handled."""
+
+    with patch(
+        "bluecurrent_api.Client.validate_api_token",
+        side_effect=WebsocketError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={"api_token": "123"},
+        )
+        assert result["errors"] == {"base": "cannot_connect"}
```

Comment:
For these tests, please checkout `@pytest.mark.parametrize()`, you can easily add more test cases when the logic of the test is the same, checkout the LastFM config flow tests.

Also, please make these error tests end in either an abort or create entry, this way we also test if the flow is able to recover from a failure.